responsemanager/responseassembler/transactionBuilder.go

```diff
@@ -0,0 +1,147 @@
+package responseassembler
+
+import (
+	blocks "github.com/ipfs/go-block-format"
+	logging "github.com/ipfs/go-log"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+
+	"github.com/ipfs/go-graphsync"
+	gsmsg "github.com/ipfs/go-graphsync/message"
+	"github.com/ipfs/go-graphsync/notifications"
+)
+
+var log = logging.Logger("graphsync")
+
+type responseOperation interface {
+	build(responseBuilder *gsmsg.Builder)
+	size() uint64
+}
+
+type transactionBuilder struct {
+	requestID   graphsync.RequestID
+	operations  []responseOperation
+	notifees    []notifications.Notifee
+	linkTracker *peerLinkTracker
+}
+
+func (prts *transactionBuilder) SendResponse(link ipld.Link, data []byte) graphsync.BlockData {
```

Comment:
I guess these methods are only public because they implement a public interface. So I guess it's ok not to comment them, but the interface methods themselves should be commented (see above).